controllers/profile.js

```diff
@@ -1,26 +1,51 @@
-const User = require("../models/User");
 const Patient = require("../models/Patient");
 const Doctor = require("../models/Doctor");
+const DateTime = require("../models/DoctorDateTime");
 
 exports.profilePatientGet = async (req, res) => {
     try {
         const patient = await Patient.findOne({
             userData: req.user.id,
-        }).populate({
-            path: "userData",
-        });
-        const userData = {
-            name: patient.userData.name,
-            lastName: patient.userData.lastName,
-            photo: patient.userData.photo,
-            phone: patient.userData.phone,
-            email: patient.userData.email,
-            visit: patient.visit,
-            birthday: patient.birthday,
-            address: patient.address,
-        };
-
-        res.status(200).json(userData);
+        })
+            .populate({
+                path: "userData",
+            })
+            .populate({
+                path: "visit",
+                populate: {
+                    path: "doctorData",
+                    select: ["direction", "workPlace", "userData"],
+
+                    populate: [
+                        { path: "workPlace", select: ["name", "address"] },
+                        { path: "userData", select: ["name", "lastName"] },
+                    ],
+                },
+            });
+        if (patient) {
+            const visits = patient.visit.map((visitItem) => ({
+                doctorName: visitItem.doctorData.userData.name,
+                doctorLastName: visitItem.doctorData.userData.lastName,
+                doctorDirection: visitItem.doctorData.direction,
+                medCenterName: visitItem.doctorData.workPlace.name,
+                medCenterAddress: visitItem.doctorData.workPlace.address,
+                time: visitItem.time,
+                date: visitItem.date,
+                _idDate: visitItem._id,
+            }));
+
+            const userData = {
+                name: patient.userData.name,
+                lastName: patient.userData.lastName,
+                photo: patient.userData.photo,
+                phone: patient.userData.phone,
+                email: patient.userData.email,
+                visit: visits,
+                birthday: patient.birthday,
+                address: patient.address,
```

Comment:
the keys of these two objects are exactly the same, what's the point of listing them?